conn.go

```diff
@@ -257,7 +257,7 @@ func (s *Session) dialWithoutObserver(ctx context.Context, host *HostInfo, cfg *
 		errorHandler:  errorHandler,
 		compressor:    cfg.Compressor,
 		session:       s,
-		streams:       streams.New(cfg.ProtoVersion),
+		streams:       s.streamIDGenerator(cfg.ProtoVersion),
```

Comment:
what if we call it `newStreamIDGenerator`?
Also is it obligatory to pass config there if we access cfg from attached type?

Reply:
Fixed, renamed and moved the function to the `streams` package.